amaru/src/sync/mod.rs

```diff
@@ -38,23 +50,33 @@ fn define_gasket_policy() -> gasket::runtime::Policy {
     }
 }
 
-pub fn bootstrap(config: Config) -> miette::Result<Vec<Tether>> {
-    let mut pull = pull::Stage::new(
-        config.upstream_peer.clone(),
-        config.network_magic,
-        config.intersection.clone(),
-    );
+pub fn bootstrap(config: Config, client: &Arc<Mutex<PeerClient>>) -> miette::Result<Vec<Tether>> {
```

Comment:
The client probably doesn't need to be Arc-Mutex as @scarmuega was saying. It can instead be destructured into individual protocols. The chainsync can be owned by the pull Stage, and the blockfetch can be owned by the validate Stage, or the fetch Stage if we separate that out.